c7n/resources/s3.py

```diff
@@ -3870,8 +3870,15 @@ def get_item_values(self, b):
 
     def augment_bucket_replication(self, b, replication, client):
         destination_bucket = replication.get('Destination').get('Bucket').split(':')[5]
+        # Ensure bucket exists
+        try:
+            client.head_bucket(Bucket=destination_bucket)
```

Comment:
We may want to use the HTTP-driven bucket check that we use to determine output bucket regions.

Reply:
Thanks for the comment. Code is updated